### PR TITLE
[#74] [Chore] Refactor to use invoke operator in Kotlin and callAsFunction in Swift

### LINKED ISSUE #74: Refactor to use `invoke` operator in Kotlin and `callAsFunction` in Swift
## Why

To make it easier to call a UseCase, we can use `invoke` operator in Kotlin and `callAsFunction` in Swift

## Who Benefits?

Developer

### PR DESCRIPTION
**Pull request by markgravity**
https://github.com/markgravity/kmm-ic/issues/74

## What happened 👀

- Refactor to use `operator func invoke` in the Kotlin UseCase
- Update the `AutoSharedUseCase.stencil` to generate the UseCase function which uses `callAsFunction` instead
- Remove redundant code in `App.swift`
## Insight 📝

n/a

## Proof Of Work 📹

CI successfully

### CODE REVIEW

File: ios/KMMIC.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -1,5 +1,14 @@
 {
   "pins" : [
+    {
+      "identity" : "nuke",
+      "kind" : "remoteSourceControl",
+      "location" : "https://github.com/kean/Nuke",
```
Comment (phongvhd93) [inline ios/KMMIC.xcworkspace/xcshareddata/swiftpm/Package.resolved]: What is the purpose of Nuke here? I believe this is unrelated to this PR since Nuke is for loading images.

Comment (markgravity) [reply ios/KMMIC.xcworkspace/xcshareddata/swiftpm/Package.resolved]: @phongvhd93 reverted it at [fd6c604](https://github.com/markgravity/kmm-ic/pull/75/commits/fd6c6044dbaf21e6dadd02ed81aa46b248f39f3f) 🙏

File: shared/src/commonTest/kotlin/co/nimblehq/mark/kmmic/domain/usecase/GetCurrentUserUseCaseTest.kt [hunk 2]
```diff
@@ -57,7 +57,7 @@ class GetCurrentUserUseCaseTest {
                 }
             )
 
-        useCase.invoke().collect {
+        useCase().collect {
             it shouldBe mockUser
         }
```
Comment (AVI5HEK) [inline shared/src/commonTest/kotlin/co/nimblehq/mark/kmmic/domain/usecase/GetCurrentUserUseCaseTest.kt]: Optional:
```suggestion
        useCase().first() shouldBe mockUser
```

**Review by AVI5HEK — APPROVED** [review]
All in all, LGTM.

**Review by suho — APPROVED** [review]

**Comment by suho** [comment]
@markgravity Please rebase the branch
